.github/workflows/release-chainhook-event-observer.yaml

```diff
@@ -0,0 +1,48 @@
+name: Build Release for Chainhook 
+on:
+  pull_request:
+  push:
+    tags:
+     - 'v*'
+    paths-ignore:
+     - '**/CHANGELOG.md'
+     - '**/package.json'
+     - '**/package-lock.json'
+    paths:
+     - components/chainhook-event-observer
+    branches:
+     - main
+     - develop
+
+  workflow_dispatch:
+
+jobs:
+  pre_run:
+    name: Cancel previous runs
+    runs-on: ubuntu-latest
+    steps:
+      - name: Cancel Previous Runs
+        uses: styfle/cancel-workflow-action@ad6cb1b847ffb509a69b745b6ee2f1d14dfe14b8
+        with:
+          access_token: ${{ github.token }}
+
+  typescript:
+    name: TypeScript
+    runs-on: ubuntu-latest
+    needs: pre_run
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v2
+
+      - name: Install stable toolchain
+        uses: actions-rs/toolchain@v1
+        with:
+          profile: minimal
+          toolchain: stable
+          override: true
+
+      - name: Publish to Crates
+        run: cargo publish --token ${CRATES_TOKEN}
+        env:
+          CRATES_TOKEN: ${{ secrets.CRATES_TOKEN }}
+        working-directory: ./components/chainhook-event-observer/
```

Comment:
When you navigate to this crate both a binary/executable and a library can be produced. In this case, we want to produce a library. Should we pass an explicit `--lib` or something?